plugins/tangle/solidifier.go

```diff
@@ -56,22 +56,26 @@ func TriggerSolidifier() {
 	milestoneSolidifierWorkerPool.TrySubmit(milestone.Index(0), true)
 }
 
-func markTransactionAsSolid(cachedTx *tangle.CachedTransaction) {
+func markTransactionAsSolid(cachedTxMeta *tangle.CachedMetadata) {
 	// Construct the complete bundle if the tail got solid (before setting solid flag => otherwise not threadsafe)
-	if cachedTx.GetTransaction().IsTail() {
-		tangle.OnTailTransactionSolid(cachedTx.Retain()) // tx pass +1
+	if cachedTxMeta.GetMetadata().IsTail() {
+		cachedTx := tangle.GetCachedTransactionOrNil(cachedTxMeta.GetMetadata().GetTxHash())
+		if cachedTx == nil {
+			log.Panicf("markTransactionAsSolid: Transaction not found: %v", cachedTxMeta.GetMetadata().GetTxHash().Trytes())
+		}
+		tangle.OnTailTransactionSolid(cachedTx) // tx pass +1
```

Comment:
OnTailTransactionSolid and then tryConstructBundle can use metadata only too as far as I can see

Reply:
hmm... tryConstructBundle is also using CurrentIndex and LastIndex, so maybe not :(

Reply:
No we need the Transaction indexes to construct the bundle, which are only stored in the transaction.
Was thinking about adding that to the metadata as well, but this function is only called when we receive a transaction, so we have the object in cache anyway.... and we will remove the bundle scam soom.